protocols/imap/src/main/java/org/apache/james/imap/api/process/ImapSession.java

```diff
@@ -33,8 +37,53 @@
  * @version $Revision: 109034 $
  */
 public interface ImapSession {
+    class SessionId {
+        public static SessionId random() {
+            return new SessionId(UUID.randomUUID());
```

Comment:
I would rather generate a random String with a SID prefix. It's cool when you want to grep some logs.

Reply:
> I would rather generate a random String with a SID prefix

I don't understand

Reply:
Do you mean something like : `SID-{UUID}`? Why not just add the prefix in the `asString` method then, if it's for the sake of grepping logs?

Reply:
because toString will only happen in debug, if you pass it, say, into a request to a service or a database, it allows to keep that type information.
We can skip this if you want but it's cheap and can help